### ISSUE TITLE
when "@ManyToOne " with "PK Auto generation" 

### ISSUE THREAD

**Issue by jinagko** [Impetus/kundera#313]
I had use two Entites with "PK Auto generation" 
1.Yh_info Entity

<pre><code>
@Id
@Column(name = "UserID")
@TableGenerator(name = "id_gen", allocationSize = 1, initialValue = 1)
@GeneratedValue(generator = "id_gen", strategy = GenerationType.TABLE)
private Long userid;
......
@ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
@JoinColumn(name = "regionid")
private Jg_region jg_region;
......
</code></pre>

2.Jg_region Entity

<pre><code>
@Id
@Column(name = "RegionID")
@TableGenerator(name = "id_gen", allocationSize = 1, initialValue = 1)
@GeneratedValue(generator = "id_gen", strategy = GenerationType.TABLE)
private Long regionid;
.....
</code></pre>

problems:
   when I create new Object Yh_info

<pre><code>Yh_info yh_info = new Yh_info();
Jg_region jg_region = new Jg_region();
jg_region.setRegionid(1);
yh_info.setJg_region(jg_regionid);
em.persist(yh_info);
 </code></pre>

when I run this tiwce or more, I expected the "regionid" is "1",but the new Yh_info.Jg_region.regionid is increase
;
It means I  assign the regionid value is "1" , but it auto increase  to "2";

Maybe it means what I want to express.
In this, "empAdr_result " should have merge in Address Equals 1 ; but when EmployeeInfo is persisted ,EmployeeAddress persist too;

  <pre><code> EmployeeAddress address_info = new EmployeeAddress();
        address_info.setStreet("street");
    em.persist(address_info);
    EmployeeAddress empAdr_result = em.find(EmployeeAddress.class, 1l);
    Assert.assertEquals("1", empAdr_result.getAddress().toString());

```
EmployeeInfo emp_info = new EmployeeInfo();
emp_info.setAddress(empAdr_result);
    em.persist(emp_info);
empAdr_result = em.find(EmployeeAddress.class, 2l);
Assert.assertEquals("1", empAdr_result.getAddress().toString());
Assert.assertEquals("2", empAdr_result.getAddress().toString());</code></pre>
```

I guest it is persist(jg_region) every times;I just want to merge(jg_region) ;

I means it should not have generate a new EmployeeAddress which Address Equals 2 , it should have always merge into EmployeeAddress which Address Equals 1;

<pre><code>
 EmployeeInfo emp_info_result =  em.find(EmployeeInfo .class, 1l);
Assert.assertEquals("1", emp_info_result .getAddress().getAddress().toString());
Assert.assertEquals("2", emp_info_result .getAddress().getAddress().toString());</code></pre>

It's so kind of your to read this,Could you give me a help~

**Comment by mevivs** [comment]
Could you please describe issues which you are getting into?

-Vivek

**Comment by jinagko** [comment]
I'm sorry,accident Enter BUTTON,editing NOW!

**Comment by mevivs** [comment]
Any entity containing only row key will not get persisted in Cassandra. Cassandra does not allows this.
Jg_region Entity is containing any other non row key column?

-Vivek

**Comment by mevivs** [comment]
Jg_region must contain a non row key column, hence it will not get persisted. As Cassandra does not store it down with row key value only.

-Vivek

**Comment by jinagko** [comment]
I have other colunn; the problem is  I expect meger the jg_region in id=1 ; but it persist a new object

**Comment by mevivs** [comment]
Hi,

```
I have other colunn; the problem is I expect meger the jg_region in id=1 ; but it persist a new object
```

Can you please elaborate more on this?

Also, as per

```
Yh_info yh_info = new Yh_info();
Jg_region jg_region = new Jg_region();
jg_region.setRegionid(1);
yh_info.setJg_region(jg_regionid);
em.persist(yh_info);
```

i can't see adding any non row key column in "jg_region" entity.
-Vivek

**Comment by mevivs** [comment]
Hi,
Just added a test case to demonstrate this. Please have a look at:
https://github.com/impetus-opensource/Kundera/blob/trunk/kundera-cassandra/src/test/java/com/impetus/client/generatedId/entites/EmployeeInfoTest.java

**Comment by jinagko** [comment]
Hi,
forgive my poor English~

I have been off work，I type this code in Text-Editor;

 Maybe it means what I want to express.
In this, "empAdr_result "  should  have merge in Address Equals 1 ;  but when EmployeeInfo is persisted ,EmployeeAddress persist too; 

<pre><code>   EmployeeAddress address_info = new EmployeeAddress();
        address_info.setStreet("street");
    em.persist(address_info);
    EmployeeAddress empAdr_result = em.find(EmployeeAddress.class, 1l);
    Assert.assertEquals("1", empAdr_result.getAddress().toString());
    EmployeeInfo emp_info = new EmployeeInfo();
    emp_info.setAddress(empAdr_result);
        em.persist(emp_info);
    empAdr_result = em.find(EmployeeAddress.class, 2l);
    Assert.assertEquals("1", empAdr_result.getAddress().toString());
    Assert.assertEquals("2", empAdr_result.getAddress().toString());</code></pre>

**Comment by jinagko** [comment]
I means it should not have generate a new EmployeeAddress which Address Equals 2 , it should have always merge into EmployeeAddress  which Address Equals 1; 

<pre><code> EmployeeInfo emp_info_result =  em.find(EmployeeInfo .class, 1l);
Assert.assertEquals("1", emp_info_result .getAddress().getAddress().toString());
Assert.assertEquals("2", emp_info_result .getAddress().getAddress().toString());</code></pre>

**Comment by mevivs** [comment]
Fixed and releasing with 2.6. Please verify.
-Vivek
